src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -226,11 +218,7 @@ function source_terms_standard(u, x, t, equations::IdealMhdMultiIonEquations2D)
     s4 = r_rho * (v1_diff * B2 - v2_diff - B1)
     s5 = v1 * s2 + v2 * s3 + v3 * s4
 
-    s[3 + (k - 1) * 5 + 1] = 0
-    s[3 + (k - 1) * 5 + 2] = s2
-    s[3 + (k - 1) * 5 + 3] = s3
-    s[3 + (k - 1) * 5 + 4] = s4
-    s[3 + (k - 1) * 5 + 5] = s5
+    set_component!(s, k, 0, s2, s3, s4, s5, equations)
```

Comment:
@amrueda You asked [here](https://github.com/amrueda/Trixi.jl/commit/57b8ea426a9b37113fb3d73088e720afa88a82dd#r112342833):
> What about here, @sloede? Should I use 0.0?

No, here `0` is correct. The thinking is that every number-like type should understand what it means to assign a literal `0` to an entry of  `s`, while `0.0` would require a conversion from `Float64` to whatever type is used as the element type of `s`.

Reply:
Note: I am answering here since this means the conversation becomes part of the PR (and not just part of the commit)